### PR TITLE
Stop using deprecated call to FindObjectsOfType with Unity 23.1+ (ABC-391)

### PR DESCRIPTION
**Pull request by thomas-tu**
## Purpose of this PR

**Jira #:** [ABC-391](https://jira.unity3d.com/browse/ABC-391)

This PR applies some API changes that happens on trunk since 23.1. It was throwing warnings in the Console when the package is installed.

## Testing

**Functional Testing status:** Manually tested by re-installing the package and making sure we don't get warnings.

**Performance Testing status:** N/A

## Documentation & UX Writing

| User facing text to review |  | Details |
| :--- | :--- | :--- |
| User interface | N |  |
| Public API docs | N |  |
| Changelog | Y |  |

| Documentation halo effect |  | Jira link |
| :--- | :--- | :--- |
| User manual | N |  |
| Other docs | N |  |

## Additional information

**Note to reviewers:**

**Reminder:**

### CODE REVIEW

**Review by cguer — APPROVED** [review]
QA approved 👍!

Tested on:
- 2023.2.0a14 (Windows only)

File: com.unity.formats.alembic/Runtime/Scripts/Analytics/AlembicExporterAnalytics.cs [hunk 1]
```diff
@@ -53,8 +53,11 @@ internal static AlembicExporterAnalyticsEvent CreateEvent(AlembicRecorderSetting
             {
                 return settings.TargetBranch != null ? settings.TargetBranch.GetComponentsInChildren<T>() : Enumerable.Empty<T>();
             }
-
+#if UNITY_2023_1_OR_NEWER
+            return Object.FindObjectsByType<T>(FindObjectsSortMode.InstanceID);
```
Comment (RoseHirigoyen) [inline com.unity.formats.alembic/Runtime/Scripts/Analytics/AlembicExporterAnalytics.cs:57]: Why InstanceID instead of None?

**Review by RoseHirigoyen — APPROVED** [review]

File: com.unity.formats.alembic/Runtime/Scripts/Analytics/AlembicExporterAnalytics.cs (hunk 1 above)
Comment (thomas-tu) [reply com.unity.formats.alembic/Runtime/Scripts/Analytics/AlembicExporterAnalytics.cs:57]: So it matches the default behavior.

> Obsolete: This function is obsolete, use [Object.FindObjectsByType](https://docs.unity3d.com/2023.2/Documentation/ScriptReference/Object.FindObjectsByType.html) instead. This replacement allows you to specify whether to sort the resulting array. FindObjectsOfType() always sorts by InstanceID, so calling FindObjectsByType(FindObjectsSortMode.InstanceID) produces identical results. If you specify not to sort the array, the function runs significantly faster, however, the order of the results can change between calls.

https://docs.unity3d.com/2023.2/Documentation/ScriptReference/Object.FindObjectsOfType.html

**Review by windxu88 — APPROVED** [review]

**Review by lucillecaillaud — APPROVED** [review]

File: com.unity.formats.alembic/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ All notable changes to this package will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+### Fixed
+- Prevent warnings related to FindObjectOfType<> calls on package installation with Unity 23.1+.
```
Comment (sebastienduverne) [inline com.unity.formats.alembic/CHANGELOG.md:9]:
```suggestion
- Prevent warnings related to FindObjectOfType<> calls on package installation with Unity Editor 2023.1+.
```

**Review by sebastienduverne — APPROVED** [review]
Just a detail.

**Pull request merged**